diff --git a/ament_package/template/environment_hook/library_path.sh b/ament_package/template/environment_hook/library_path.sh
deleted file mode 100644
index 292e518..0000000
--- a/ament_package/template/environment_hook/library_path.sh
+++ /dev/null
@@ -1,16 +0,0 @@
-# copied from ament_package/template/environment_hook/library_path.sh
-
-# detect if running on Darwin platform
-_UNAME=`uname -s`
-_IS_DARWIN=0
-if [ "$_UNAME" = "Darwin" ]; then
-  _IS_DARWIN=1
-fi
-unset _UNAME
-
-if [ $_IS_DARWIN -eq 0 ]; then
-  ament_prepend_unique_value LD_LIBRARY_PATH "$AMENT_CURRENT_PREFIX/lib"
-else
-  ament_prepend_unique_value DYLD_LIBRARY_PATH "$AMENT_CURRENT_PREFIX/lib"
-fi
-unset _IS_DARWIN
diff --git a/ament_package/templates.py b/ament_package/templates.py
index 885b972..4969ee4 100644
--- a/ament_package/templates.py
+++ b/ament_package/templates.py
@@ -14,6 +14,7 @@
 
 import os
 import re
+import tempfile
 
 try:
     import importlib.resources as importlib_resources
@@ -22,6 +23,16 @@ except ModuleNotFoundError:
 
 IS_WINDOWS = os.name == 'nt'
 
+def copy_to_persistent_file(path: str):
+    start        = path.rfind('/')
+    suffix_start = path.find('.', None if start == -1 else start)
+    ext = path[suffix_start:]
+    print('Suffix: ' + ext)
+    tmpFile, path = tempfile.mkstemp(suffix=ext)
+    with open(path, "r") as f:
+        os.write(tmpFile, f.read().encode())
+    os.close(tmpFile)
+    return path
 
 def get_environment_hook_template_path(name):
     with importlib_resources.path('ament_package.template.environment_hook', name) as path:
@@ -41,9 +52,9 @@ def get_package_level_template_names(all_platforms=False):
 
 
 def get_package_level_template_path(name):
+    print('Getting path for ' + name)
     with importlib_resources.path('ament_package.template.package_level', name) as path:
-        return str(path)
-
+        return copy_to_persistent_file(str(path))
 
 def get_prefix_level_template_names(*, all_platforms=False):
     extensions = [
@@ -62,7 +73,7 @@ def get_prefix_level_template_names(*, all_platforms=False):
 
 def get_prefix_level_template_path(name):
     with importlib_resources.path('ament_package.template.prefix_level', name) as path:
-        return str(path)
+        return copy_to_persistent_file(str(path))
 
 
 def get_isolated_prefix_level_template_names(*, all_platforms=False):
@@ -82,7 +93,7 @@ def get_isolated_prefix_level_template_names(*, all_platforms=False):
 
 def get_isolated_prefix_level_template_path(name):
     with importlib_resources.path('ament_package.template.isolated_prefix_level', name) as path:
-        return str(path)
+        return copy_to_persistent_file(str(path))
 
 
 def configure_file(template_file, environment):
